### PR TITLE
Avago3008 update v5

### PR DESCRIPTION
**Pull request by vrbagalkote**
Attached is the JOB.log o/p.
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/621252/job.txt)

### CODE REVIEW

File: io/disk/Avago_storage_adapter/avago3008.py [hunk 1]
```diff
@@ -40,47 +40,140 @@ def setUp(self):
...
+                self.skip("Raid1e needs minimum of 3 disks to be tested")
+            else:
+                self.raid_disk = " ".join(self.disk).strip(" ")
+        if self.raidlevel == 'raid10':
+            if self.number_of_disk < 4:
+                self.skip("Raid10 needs minimum of 4 disks to be tested")
+            else:
+                self.raid_disk = " ".join(self.disk).strip(" ")
+        if self.raidlevel == 'raid0':
+            self.raid_disk = " ".join(self.disk).strip(" ")
```
Comment (narasimhan-v) [inline io/disk/Avago_storage_adapter/avago3008.py]: Please try to optimize the common logic here.

File: io/disk/Avago_storage_adapter/avago3008.py [hunk 2]
```diff
@@ -40,47 +40,140 @@ def setUp(self):
...
-        This function creates raid1 array
+        This function waits, till the current operation is complete
         """
+        while current_operation != 'None':
+            time.sleep(10)
+            self.adapter_status()
+            cmd = "./sas3ircu 0 status | grep 'Current operation' | \
+                 awk '{print $4}'"
+            current_operation = process.system_output(cmd, shell=True) \
+                .strip("\n")
```
Comment (narasimhan-v) [inline io/disk/Avago_storage_adapter/avago3008.py]:
```
cmd = "./sas3ircu 0 status | grep 'Current operation' | awk '{print $4}'"
while process.system_output(cmd, shell=True).strip("\n") != 'None':
    time.sleep(10)
```

File: io/disk/Avago_storage_adapter/avago3008.py [hunk 3]
```diff
@@ -40,47 +40,140 @@ def setUp(self):
...
+    def rebuild(self):
+
+        """
+        This functions waits for the rebuild to complete on a Raid
+        """
+        self.set_online_offline("offline")
+        self.status = "degraded"
+        while self.status != 'Optimal':
+            time.sleep(30)
+            self.status = self.adapter_status().strip("\n")
```
Comment (narasimhan-v) [inline io/disk/Avago_storage_adapter/avago3008.py]: can be done similar to sleepfunction()

File: io/disk/Avago_storage_adapter/avago3008.py [hunk 4]
```diff
@@ -92,42 +185,74 @@ def set_online_offline(self, state):
...
+        This is a helper function, to check the status of the adapter
         """
-        self.set_online_offline("offline")
+        cmd = "./sas3ircu %d status" % self.controller
+        output = process.run(cmd, shell=True, ignore_status=True)
+        if output.exit_status != 0:
+            self.fail("Failed to display the status of the adapter")
+        for i in output.stdout.splitlines():
+            if 'Volume state' in i:
+                return i.split()[-1]
```
Comment (narasimhan-v) [inline io/disk/Avago_storage_adapter/avago3008.py]: Suggestion: you can overload this function to get either 'Vol State' or 'Curr Operation'

File: io/disk/Avago_storage_adapter/avago3008.py [hunk 5]
```diff
@@ -92,42 +185,74 @@ def set_online_offline(self, state):
...
+            self.fail("Failed to create raid on the drives")
+
+    def deleteraid(self):
+
+        """
+        This function deletes raid array
+        """
+        cmd = "./sas3ircu %d display | grep 'vr1' -B 2 | grep 'Volume ID' | \
+               awk '{print $4}'" % self.controller
+        volume_id = int(process.system_output(cmd, shell=True))
```
Comment (narasimhan-v) [inline io/disk/Avago_storage_adapter/avago3008.py]: Can become a helper function

Comment (vrbagalkote) [inline io/disk/Avago_storage_adapter/avago3008.py]: Done!!

File: io/disk/Avago_storage_adapter/avago3008.py (hunk 3 above)
Comment (vrbagalkote) [inline io/disk/Avago_storage_adapter/avago3008.py]: Done!!

File: io/disk/Avago_storage_adapter/avago3008.py (hunk 2 above)
Comment (vrbagalkote) [inline io/disk/Avago_storage_adapter/avago3008.py]: Done!!

File: io/disk/Avago_storage_adapter/avago3008.py (hunk 4 above)
Comment (vrbagalkote) [inline io/disk/Avago_storage_adapter/avago3008.py]: Done!!

File: io/disk/Avago_storage_adapter/avago3008.py (hunk 1 above)
Comment (vrbagalkote) [inline io/disk/Avago_storage_adapter/avago3008.py]: Done!!!

**Comment by vrbagalkote** [comment]
Addressed all the comments and raised a new PR.

https://github.com/avocado-framework-tests/avocado-misc-tests/pull/193

**Pull request closed without merging**